booking-service/src/test/java/fr/xebia/xke/micronaut/booking/StockTest.java

```diff
@@ -40,17 +40,33 @@ void should_fail_to_add_null_quantity() {
         final Stock initialStock = Stock.of(ARTICLE, 12);
 
         assertThatNullPointerException()
-                .isThrownBy(() -> initialStock.add(null));
+                .isThrownBy(() -> initialStock.subtract(null));
     }
 
     @Test
-    void should_add_quantity() {
+    void should_subtract_quantity() {
         final Stock initialStock = Stock.of(ARTICLE, 12);
-        final Quantity increment = new Quantity(1L);
-        final Stock incrementedStock = Stock.of(ARTICLE, 13);
+        final Quantity decrement = new Quantity(1L);
+        final Stock decrementedStock = Stock.of(ARTICLE, 11);
 
-        assertThat(initialStock.add(increment))
-                .isEqualTo(incrementedStock);
+        assertThat(initialStock.subtract(decrement))
+                .isEqualTo(decrementedStock);
     }
 
+    @Test
+    void should_fail_to_subtract_unavailable_quantity() {
+        final Stock initialStock = Stock.of(ARTICLE, 12);
+        final Quantity decrement = new Quantity(13L);
+
+        assertThatExceptionOfType(UnavailableArticleQuantityException.class)
+                .isThrownBy(() -> initialStock.subtract(decrement));
```

Comment:
you can also test the error message by calling `.withMessage("Requested quantity (13) is unavailable for article 'BOOK30004' (available: 12)")`